### PR TITLE
Clean up skybox rendering

### PR DESCRIPTION
**Pull request by zzmp**
- Allow procedural or color-only rendering without a specified texture
- Expose color to the skybox procedural shader
- Properly detect if/when textures are loaded

### CODE REVIEW

**Comment by zzmp** [comment]
https://app.asana.com/0/74051501495175/96203467746694
https://app.asana.com/0/74051501495175/95957371936708

#### Intents

Properly load skymaps. When switching between them, they would attempt to load after the data was "loaded" but before it was "read", so that it would fail and stop trying.

#### Testing

Ensure that switching between many, uncached skymaps works. There should be a new debug message. Along with "Starting request for", "Request finished for", and "Finished loading", there should be a new "Finished reading" specific to loading textures.

Procedural and color-only skyboxs should now work without a texture specified.

Bad skybox textures should continue to not crash anything.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad74a1d52911cc95bfe363439e58df3c1d2a3450 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by OmegaHeron** [comment]
Testing on Windows 7 - Using Sunset Skybox Zone from Marketplace - it's not displaying sky texture

hifi-snap-by-omegaheron-on-2016-03-09_10-37-25

Log shows

`[03/09 10:37:12] [DEBUG] Starting request for: https://hifi-public.s3.amazonaws.com/images/SkyboxTextures/SunSet.jpg`
`[03/09 10:37:12] [DEBUG] Request finished for https://hifi-public.s3.amazonaws.com/images/SkyboxTextures/SunSet.jpg`
`[03/09 10:37:12] [DEBUG] Finished loading: https://hifi-public.s3.amazonaws.com/images/SkyboxTextures/SunSet.jpg`
`[03/09 10:37:12] [DEBUG] Cube map size: "https://hifi-public.s3.amazonaws.com/images/SkyboxTextures/SunSet.jpg" 2048 12288`
`[03/09 10:37:17] [DEBUG] Finished reading: https://hifi-public.s3.amazonaws.com/images/SkyboxTextures/SunSet.jpg`

Differences in this zone from Marketplace version is it doesn't fill full domain
It's located at 0,0,0 with size = 4096,4096,4096
Stage Sun Model Enabled = Yes
No ambient URL set
Automatic stage hour/day = Yes

Reverting to same Zone as dropped in from MP leads to same thing - black sky.

**Comment by OmegaHeron** [comment]
Same at hifi://sandbox

hifi-snap-by-omegaheron-on-2016-03-09_10-44-44

**Comment by RyanDowne** [comment]
I can confirm that when I remove the skybox url, I do get the skybox color so that's nice. But, I am still seeing inconsistent skybox texture loading issues. 

Procedural skyboxes seem to work but if you go to 'sandbox' or 'demo' they are only using a skybox texture and those aren't displaying for me on my PC.

**Comment by zzmp** [comment]
This will need to be rebased on #7292 when it lands. DNM until then, and I'll investigate how I broke the skybox textures. Thanks for the tests, guys!

**Comment by zzmp** [comment]
Need to test myself before I take away DNM.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b627a17ce911974bcb72ebeb7c24fc5c10abbf5a available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by zzmp** [comment]
I degraded the procedural shader feature by taking out access to the texture if it is there.
I am fixing that, writing a test script, and then putting this up for QA.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ade9ff9f223b473f4e5ea466459cb649328f4781 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1ec87d253d26151557f9a7e58bd4237621bc6f76 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3acf27a027c22e4b8ef725dfccc188839616e4c1 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 39f444f6a62d2c77c68fbd185b7d5a5d0e668b4d available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: beab9bef07a69e46a5b18a80ee9dcaa3f8529276 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a11fb5bb3b0b1c638a7208028638ad32b3f93e9a available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 85ec4385063731151f722a075a3c60130a9f1bdc available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2508b1412680aedffa7b0deb17d8e34ebe4e2fed available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 755efd1840ddac2b516706f6fd3943c9951537d1 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 755efd1840ddac2b516706f6fd3943c9951537d1 available at: https://gist.github.com/48e931b2c81c9f8c67a3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7286.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7 - cycled through all permutations via script and seen result matches script description.  Existing skybox zone works as expected.  QA Pass

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3757,19 +3757,19 @@ namespace render {
...
             }
-                // fall through to next case
+
+            // Fall through: if no skybox is available, render the SKY_DOME
             case model::SunSkyStage::SKY_DOME:  {
                 if (Menu::getInstance()->isOptionChecked(MenuOption::Stars)) {
                     PerformanceTimer perfTimer("stars");
                     PerformanceWarning warn(Menu::getInstance()->isOptionChecked(MenuOption::PipelineWarnings),
-                        "Application::payloadRender<BackgroundRenderData>() ... stars...");
+                        "Application::payloadRender<BackgroundRenderData>() ... My god, it's full of stars...");
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: :+1:

Comment (zzmp) [inline interface/src/Application.cpp]: You found my easter egg...

**Pull request merged**